### PR TITLE
Fix : Talent level is not clamped when updating character ascension

### PR DESCRIPTION
**Pull request by notV3NOM**
## Describe your changes

 Clamp the talent level when character's ascension level is changed

## Issue or discord link

 Resolves https://github.com/frzyc/genshin-optimizer/issues/1440
 https://discord.com/channels/785153694478893126/1195110729917939773
 
## Testing/validation

Tested Locally
Areas of Impact : All weapon/character level selectors, DB Import, Random Character creation

## Checklist before requesting a review 

- [x] I have commented my code, in potentially confusing areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [x] Ran `yarn run mini-ci` locally to validate format + lint.
- [ ] If there were format issues, I ran `nx format write` to resolve them automatically.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: apps/frontend/src/app/Components/LevelSelect.tsx [hunk 1]
```diff
@@ -17,40 +17,76 @@ import CustomNumberInput, {
...
   const ascensionMaxLevels = useLow ? ascensionMaxLevelLow : ascensionMaxLevel
+  const setLevelAndAscensionWithTalentClamp = useCallback(
+    (newLevelAndAscension: {
+      level?: number
+      ascension: AscensionKey
+      talent?: ICharacterTalent
+    }) => {
+      const dispatchSettings = newLevelAndAscension
+      // If changing ascension for a Character, clamp their talent levels
+      if (talent) {
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Components/LevelSelect.tsx:51]: This should be in our db verification code too
https://github.com/frzyc/genshin-optimizer/blob/461cdc02f4b099906265f169e4089ef41e9d1951/apps/frontend/src/app/Database/DataManagers/CharacterData.ts#L50

Comment (notV3NOM) [reply apps/frontend/src/app/Components/LevelSelect.tsx:51]: Added in https://github.com/frzyc/genshin-optimizer/pull/1454/commits/ab499328f1a78d8600d0a556ce1b30a7d1cfa74c

File: apps/frontend/src/app/Components/LevelSelect.tsx [hunk 2]
```diff
@@ -51,8 +51,9 @@ export default function LevelSelect({
       if (talent) {
         const clampedTalent = { ...talent }
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Components/LevelSelect.tsx:52]: Looks like we can reuse the validateTalent function here?

Comment (notV3NOM) [reply apps/frontend/src/app/Components/LevelSelect.tsx]: Yep , added in https://github.com/frzyc/genshin-optimizer/pull/1454/commits/64c4f410f4d72a9b642f9e177aeb3b44a749901c . Skipping ternary operator to favour readability

**Comment by nguyentvan7** [comment]
https://genshin-optimizer-prs.github.io/pr/1454/frontend

**Review by nguyentvan7 — DISMISSED** [review]
Validated changing via UI and importing modified DB; both clamped talents as expected.

Thanks for adding unit tests as well

File: apps/frontend/src/app/Database/Database.test.ts [hunk 3]
```diff
@@ -702,4 +706,47 @@ describe('Database', () => {
       database.weapons.get(database.chars.get('Albedo')?.equippedWeapon)?.key
     ).toEqual('DullBlade')
   })
+
+  describe('Test character talent validation', () => {
```
Comment (frzyc) [inline apps/frontend/src/app/Database/Database.test.ts:710]: I totally appreciate you for putting in unit tests. Its definitely something that we need to do more of.
It looks like this is testing the `validateTalent` function, which should go into a `talent.test.ts` file next to `talent.ts`. This allows the unit test to test strictly what the `validateTalent` function does (which your database functions are doing in a round-about way).

**Review by frzyc — CHANGES_REQUESTED** [review]
All the logic LGTM, just a concern regarding unit test.

File: apps/frontend/src/app/Database/Database.test.ts (hunk 3 above)
Comment (notV3NOM) [reply apps/frontend/src/app/Database/Database.test.ts]: Yep, added in https://github.com/frzyc/genshin-optimizer/pull/1454/commits/0c030a3968f5570a23bae40f0fccafdffe4d69b9

**Review by frzyc — APPROVED** [review]
🚢

**Pull request merged**